src/network/NetworkManagerPrimary.cc

```diff
@@ -109,6 +110,9 @@ bool NetworkManagerPrimary::Ready() const
   return (nSecondary == this->dataPtr->config.numSecondariesExpected);
 }
 
+/// Number of iterations a secondary can move ahead.
+constexpr uint64_t kSecondaryIterations = 1000uLL;
```

Comment:
Should this be a constant/parameter of the networkPrimary class instead?

Reply:
Yes, and in fact, it may make more sense to have this value be configurable, depending on the application.

Reply:
Let me know what you think of a proposed change in a2e607b. Leaving a TO-DO to later build on top of that how to configure that parameter (later, if desired)